### ISSUE TITLE
2.15

### ISSUE THREAD

**Issue by karato** [yegor256/netbout#661]
Please, publish a new release **2.15** (or higher). Don't forget to update us here regularly (at least once a week), as explained in [this article](http://www.yegor256.com/2015/05/11/software-architect-responsibilities.html).

**Comment by karato** [comment]
@yegor256 milestone set to `3.1` (correct me if I am wrong)

**Comment by dmzaytsev** [comment]
@yegor256 please see the project report

Scope status:

* RESTful API [90%]
* search functionality [none]
* email enhancement [30%]
* UI enhancement[10%]
* current bugs fix [7%]

Issues:

* I have no information about available developers and their the task limits on this project. This complicates the planning. It would better if I had the the possible number of concurrent tasks for this project.
* hard to do a last review before merge for some tasks . We see an errors only after the release. As result we have a new ticket, a new assignment, a new PR, a new CR...  It's spend too much time.

Risks:

* 3.0 release make take more time. It's a campus project [5/6]

**Comment by yegor256** [comment]
@dmzaytsev many thanks for the report. I think that we should create a "staging" deployment of Netbout, a full copy of production. Every time we "deploy" the snapshot, this staging will be updated. It will help us to test the product before putting it to production. Something like `staging.netbout.com`. What do you think?

**Comment by yegor256** [comment]
@dmzaytsev or maybe `snapshot.netbout.com`, for clarity

**Comment by dmzaytsev** [comment]
@yegor256 in my opinion `snapshot.netbout.com` is good .
we have to merge then deploy and test on the `snapshot.netbout.com`, right ?
What we should to do if after checking we see task not completed?
The performer have to make a new PR?

**Comment by dmzaytsev** [comment]
@yegor256 in this case the `master` branch and the `netbout.com` will be different. It's not good I think. Maybe it would better to create a new branch e.g. `dev`, merge the changes into it and `snapshot.netbout.com`(or `dev.netbout.com`) will be updated from `dev`.
what do you think?

**Comment by karato** [comment]
@dmzaytsev do this task

**Comment by yegor256** [comment]
@dmzaytsev master branch should not be equal to production. Every time we release, we create a new Git tag, which is equal to production. The master should go forward and introduce new features. So, the most logical approach is to deploy master somewhere else and see how it works there. We'll have to create "snapshot" tables in DynamoDB, "snapshot" facebook/google/github accounts, etc. etc. It will be a completely similar system, but probably with access available only to our developers. Maybe it's good to protect it via HTTP Basic Auth?

**Comment by yegor256** [comment]
@dmzaytsev the `dev` branch you're suggesting is what GitFlow is suggesting (http://nvie.com/posts/a-successful-git-branching-model/), but I prefer to have it vice versa: master branch is where we develop and "release" branches/tags is where we keep stable versions

**Comment by dmzaytsev** [comment]
@yegor256 I agree,  BasicAuth will be enough.
just one question
> "snapshot" facebook/google/github accounts
what do you mean?

**Comment by yegor256** [comment]
@dmzaytsev there will be a possibility to login to `snapshot.netbout.com`. Current OAuth logins won't work, because they are attached to `www.netbout.com`. I will have to create new "apps" in facebook, google+, and github. Not a big problem, but has to be done

**Comment by yegor256** [comment]
@dmzaytsev see #681

**Comment by dmzaytsev** [comment]
@yegor256 could you please mark 2.15.1 as stable release?

**Comment by yegor256** [comment]
@dmzaytsev I did it for 2.15.1

**Comment by yegor256** [comment]
@dmzaytsev since we have a snapshot deployment, let's agree that you don't deploy to production until I approve what I see in the snapshot. OK?

**Comment by dmzaytsev** [comment]
@yegor256 
> @dmzaytsev I did it for 2.15.1
thanks!


> @dmzaytsev since we have a snapshot deployment, let's agree that you don't deploy to production until I approve what I see in the snapshot. OK?

sure, I agree, it would better

**Comment by karato** [comment]
@dmzaytsev It is closed now

**Comment by karato** [comment]
@dmzaytsev Thanks for your contribution, **32 mins** was added to your account, payment ID is `59061951`, 28 hours and 8 mins spent total. here is how the bonus is calculated: t=322, c=52, h=1293, diff=2.14.16..master. +32 added to your rating, at the moment it is: [+2615](http://www.netbout.com/b/35166?open=rating)
